### PR TITLE
ARROW-3231: [Python] Sphinx's autodoc_default_flags is now deprecated

### PR DESCRIPTION
**Pull request by kszucs**
Hopefully resolves the recent CI failures.

### CODE REVIEW

**Review by kou — APPROVED** [review]
+1
CI isn't finished yet but I think that this will fix the failure too.

**Review by kou — CHANGES_REQUESTED** [review]
CI is failed...

File: python/doc/source/conf.py [hunk 1]
```diff
@@ -63,8 +63,8 @@
 ]
 
 # Show members for classes in .. autosummary
-autodoc_default_flags = ['members', 'undoc-members', 'show-inheritance',
-                         'inherited-members']
+autodoc_default_options = ['members', 'undoc-members', 'show-inheritance',
+                           'inherited-members']
```
[the remark this answers is not in the document]
Comment (kszucs) [reply python/doc/source/conf.py]: Thanks for the update and reference! Fixing it...

**Comment by wesm** [comment]
+1

**Comment by kszucs** [comment]
Sphinx 1.8.0 raised a couple of other warning locally so pinning was quicker. Thanks for removing the unused import!

**Pull request closed without merging**